ui/analyse/src/ctrl.ts

```diff
@@ -128,6 +128,11 @@ export default class AnalyseCtrl {
     makeStudy?: typeof makeStudyCtrl,
   ) {
     this.data = opts.data;
+    if (opts.data.game.variant.key === 'standard' || opts.data.game.variant.key === 'fromPosition') {
+      const new_fen = this.returnStandardCastlingFen(opts.data.game.fen);
+      opts.data.game.fen = new_fen;
+      opts.data.treeParts[0].fen = new_fen;
```

Comment:
overwriting the fen coming from the server... Maybe it should be fixed on the server side instead, then?